### DISCUSSION TITLE
RTMDet

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmdetection#9713] [General]
我想问一下，RTMDet的参数量比yolov5大约少4M个，但同一张显卡，yolov5的bs能设置到256，RTMDet只能64，导致同样的数据集，训练时间多4倍，这是为啥？RTMDet的图像预处理更吃显存导致的吗？

**Comment by imAzhou** [comment]
作为提问人我贴一下配置【提问时使用的老版本，现在用的是0.4中fast版本】
显卡：一张，48G NVIDIA A40
数据集：COCO2017

YOLOV5-s：(基本是默认配置)
config: yolov5_s-v61_syncbn_8xb16-300e_coco.py
train_num_workers: 8
bs: 128
lr: 0.01
预计需要的训练时间: 3 days, 7:23:53

RTMDet-s：(bs与默认配置相比缩小了4倍，所以把学习率也调低了4倍)
config: rtmdet_s_syncbn_fast_8xb32-300e_coco.py
train_num_workers: 8
bs: 64
lr: 0.01
预计需要的训练时间: 5 days, 13:54:37

**Comment by qwert31639** [comment]
> 我想问一下，RTMDet的参数量比yolov5大约少4M个，但同一张显卡，yolov5的bs能设置到256，RTMDet只能64，导致同样的数据集，训练时间多4倍，这是为啥？RTMDet的图像预处理更吃显存导致的吗？

我認為是assigners的問題，不知道您後續有解決了嗎
https://github.com/open-mmlab/mmdetection/blob/61dd8d518b13c7ee4bdf609595b7e803f3ac0224/mmdet/models/task_modules/assigners/dynamic_soft_label_assigner.py
